src/Strategies/CartFixedDiscountStrategy.php

```diff
@@ -4,18 +4,34 @@
 
 use EscolaLms\Vouchers\Models\Cart;
 use EscolaLms\Vouchers\Models\CartItem;
+use EscolaLms\Vouchers\Services\Contracts\CouponServiceContract;
 use EscolaLms\Vouchers\Strategies\Abstracts\DiscountStrategy;
 use EscolaLms\Vouchers\Strategies\Contracts\DiscountStrategyContract;
 
 class CartFixedDiscountStrategy extends DiscountStrategy implements DiscountStrategyContract
 {
     public function calculateAdditionalDiscount(Cart $cart): int
     {
-        return $cart->totalPreDiscount < $this->coupon->amount ? $cart->totalPreDiscount : $this->coupon->amount;
+        return 0;
     }
 
     public function calculateDiscountForItem(Cart $cart, CartItem $cartItem): int
     {
-        return 0;
+        if (app(CouponServiceContract::class)->cartItemIsExcludedFromCoupon($this->coupon, $cartItem)) {
+            return 0;
+        }
+
+        $totalAmount = 0;
+        $maxAmount = $this->coupon->amount;
+
+        foreach ($cart->items as $item) {
+            if (!app(CouponServiceContract::class)->cartItemIsExcludedFromCoupon($this->coupon, $item)) {
+                $totalAmount += $item->basePrice;
+            }
+        }
+
+        $maxAmount = min($totalAmount, $maxAmount);
+
+        return round($cartItem->basePrice / $totalAmount * $maxAmount);
```

Comment:
Is there any chance that $totalAmount will be 0 at this line?

Reply:
probably when all cart items will be free